pyproject.toml

```diff
@@ -11,6 +11,7 @@ classifiers = [
 ]
 dependencies = [
     "pyarrow>=8.0.0",
+    "typing-extensions >=4.0.0,< 5.0.0",
```

Comment:
we don't need it for python 3.10+ so let's put `"typing-extensions>=4.6.1; python_version < '3.10'"`

and do in `__init__.py`

```py
if sys.version_info < (3, 10):
    from typing_extensions import TypeAlias
else:
    from typing import TypeAlias
```

Reply:
Maybe we can even avoid the dependency if we put everything under the `if TYPE_CHECKING` but I guess it's not worth it. WDYT @lukapeschke?

Reply:
Yes letting `>=4` is ok sorry

Reply:
Right at you!